File: watcomc/inp.pas
(* inp.pas -- Pascal declarations for inp.asm.

   This file is for linking compiled object files with Pascal linker.
   It will be deleted in future when we rewrite the project in C.

   This is free and unencumbered software released into the public domain.
   For more information, please refer to <http://unlicense.org>. *)

unit inp;

interface

procedure inp_;

implementation

(*$l inp.obj*)

procedure inp_; external;

end.

File: src/cc/conio/inp.c
/* inp.c -- part of custom "conio" library.

   This is free and unencumbered software released into the public domain.
   For more information, please refer to <http://unlicense.org>. */

#include "defines.h"

#ifdef __WATCOMC__
#pragma aux default "$cc$conio$inp$*"
#endif  /* __WATCOMC__ */

#include "cc/conio.h"

#ifndef __WATCOMC__

char cc_inp(unsigned int port)
{
    _asm
    {
        "mov dx,port";
        "in al,dx";
    }
}

#endif  /* !__WATCOMC__ */
